### ISSUE TITLE
FlangOptimizerTests FAILs to link on Solaris

### ISSUE THREAD

**Issue by rorth** [llvm/llvm-project#53690]
When building LLVM 14.0.0 rc1 on Solaris, I get a link failure:
```
FAILED: tools/flang/unittests/Optimizer/FlangOptimizerTests
[...]
Undefined                       first referenced
 symbol                             in file
_ZN3fir7runtimeL8getModelIcEEPFN4mlir4TypeEPNS2_11MLIRContextEEv lib/libFIRBuilder.a(Reduction.cpp.o)
ld: fatal: symbol referencing errors
```
This is `mlir::Type (*fir::runtime::getModel<char>())(mlir::MLIRContext*)` and it's used in `flang/lib/Optimizer/Builder/Runtime/Reduction.cpp`, but defined nowhere AFAICT.  This doesn't happen on Linux/x86_64, though.

**Comment by rorth** [comment]
I just came to a similar solution: patch posted at [D119438](https://reviews.llvm.org/D119438).

**Comment by tstellar** [comment]
Merged: 3001b0d51963
